jax/_src/scipy/linalg.py

```diff
@@ -289,28 +289,20 @@ def _calc_P_Q(A):
   A_L1 = np_linalg.norm(A,1)
   n_squarings = 0
   if A.dtype == 'float64' or A.dtype == 'complex128':
-   U3, V3 = _pade3(A)
-   U5, V5 = _pade5(A)
-   U7, V7 = _pade7(A)
-   U9, V9 = _pade9(A)
    maxnorm = 5.371920351148152
    n_squarings = jnp.maximum(0, jnp.floor(jnp.log2(A_L1 / maxnorm)))
    A = A / 2**n_squarings
-   U13, V13 = _pade13(A)
    conds = jnp.array([1.495585217958292e-002, 2.539398330063230e-001,
-                      9.504178996162932e-001, 2.097847961257068e+000])
-   U = jnp.select((A_L1 < conds), (U3, U5, U7, U9), U13)
-   V = jnp.select((A_L1 < conds), (V3, V5, V7, V9), V13)
+                      9.504178996162932e-001, 2.097847961257068e+000, jnp.inf])
+   idx = jnp.argwhere(A_L1 < conds, size=1).squeeze()
```

Comment:
Ah, this is your use case. Then you might also want to try jnp.digitize.

Reply:
Nice yes that is cleaner.

Reply:
But be aware that `jnp.digitize` is `searchsorted`-based, which could have different performance characteristics from the `nonzero`-based `jnp.argwhere`. I have absolutely no idea which one should/would be faster in this particular case, only through benchmark can one tell I guess.

Reply:
Ah I see - thanks. My naive expectation is that the matrix operations would dominate in this function anyway, especially given how few options either `digitize` or `argwhere` need to choose from.

Reply:
I think this would give the same result, and be more efficient than either `argwhere` or `digitize`:
```
idx = jnp.argmax(A_L1 < conds)
```

Reply:
Okay so I've just run the following code:

```
conds = jnp.array([1.495585217958292e-002, 2.539398330063230e-001,
                      9.504178996162932e-001, 2.097847961257068e+000, jnp.inf])
conds2 = jnp.array([1.495585217958292e-002, 2.539398330063230e-001,
                      9.504178996162932e-001, 2.097847961257068e+000])
```
and in different cells, for various values of `x`, I'm seeing:
```
%timeit jnp.argmax(x < conds).block_until_ready()
```
CPU (local): 178 microseconds
GPU (colab): 610 microseconds
```
%timeit jnp.argwhere(x < conds, size=1).squeeze().block_until_ready()
```
CPU (local): 1.89 ms
GPU (colab): 5.09 ms
```
%timeit jnp.digitize(x, conds2).block_until_ready()
```
CPU (local): 5.8 microseconds
GPU (colab): 270 microseconds

The above values were for `x=3.` but the speed rankings seem to be the same regardless of `x`. Seems like for this case `digitize` is the clear winner.

Going to revert this branch back to the digitize version - let me know if you think there are any issues with this.

Reply:
OK, sounds good!